app/services/middleware_topology_service.rb

```diff
@@ -1,30 +1,27 @@
 class MiddlewareTopologyService < TopologyService
+  include UiServiceMixin
```

Comment:
I did almost the same changes in my PR to support the fa glyphs. Currently, the `middleware_topology_controller.js` uses this `"/assets/svg/" + class_name(d) + ".svg"` as icon path. When using the `UiServiceMixin` we can be more generic and have some icons as glyphs and some as svgs. Small drawback may be that the "icons part of the JSON" (localhost:3000/middleware_topology/data) is growing even if not all declared icons are being used.

@ammendonca are you going to do changes also to `middleware_topology_controller.js` ?